src/main/scala/org/ergoplatform/settings/Parameters.scala

```diff
@@ -1,30 +1,283 @@
 package org.ergoplatform.settings
 
+import com.google.common.primitives.Ints
+import io.circe.Encoder
+import io.circe.syntax._
+import org.ergoplatform.api.ApiCodecs
+import org.ergoplatform.modifiers.history.{Extension, ExtensionCandidate}
+import org.ergoplatform.nodeView.history.ErgoHistory.Height
+import scorex.core.serialization.Serializer
+
+import scala.util.Try
+
 
 /**
   * System parameters which could be readjusted via collective miners decision.
   */
-object Parameters {
-
-  // Max total computation cost of a block.
-  val MaxBlockCost: Long = 1000000
+class Parameters(val height: Height, val parametersTable: Map[Byte, Int]) {
 
-  // Max size of transactions section of a block.
-  val MaxBlockSize: Int = 512 * 1024
+  import Parameters._
 
-  val Kdefault = 1250000
-  val Kmax = 2500000
-  val Kmin = 0
-  val Kstep = 50000
-
-  /** Cost of storing 1 byte per Constants.StoragePeriod blocks, in nanoErgs
+  /**
+    * Cost of storing 1 byte per Constants.StoragePeriod blocks, in nanoErgs.
     */
-  val K: Long = Kdefault
+  lazy val storageFeeFactor: Int = parametersTable(StorageFeeFactorIncrease)
 
   /** To prevent creation of dust which is not profitable to charge storage fee from, we have this min-value per-byte
     * parameter.
     */
-  val MinValuePerByte: Long = 30 * 12
+  lazy val minValuePerByte: Int = parametersTable(MinValuePerByteIncrease)
+
+  /**
+    * Max size of transactions section of a block.
+    */
+  lazy val maxBlockSize: Int = parametersTable(MaxBlockSizeIncrease)
+
+  /**
+    * Max total computation cost of a block.
+    */
+  lazy val maxBlockCost: Long = parametersTable(MaxBlockCostIncrease)
+
+  lazy val softForkStartingHeight: Option[Height] = parametersTable.get(SoftForkStartingHeight)
+  lazy val softForkVotesCollected: Option[Int] = parametersTable.get(SoftForkVotesCollected)
+
+  lazy val blockVersion: Byte = parametersTable(BlockVersion).toByte
+
+  def update(height: Height, forkVote: Boolean, epochVotes: Seq[(Byte, Int)], votingSettings: VotingSettings): Parameters = {
+    val table1 = updateFork(height, parametersTable, forkVote, epochVotes, votingSettings)
+    val table2 = updateParams(table1, epochVotes, votingSettings.votingLength)
+    Parameters(height, table2)
+  }
+
+  def updateFork(height: Height,
```

Comment:
This method conditions are hard to read and are not described in yellow paper. This looks like a state machine with multiple possible transitions, but after spending 30 minutes reviewing this method it is still unclear for me, that it is not possible to stuck at some state (e.g. what if we'll come to `successful voting - cleaning after activation` state while `votes <= votingEpochLength * votingEpochs * 9 / 10`?)
Let's add human-readable description of possible transactions to yellow paper.